### DISCUSSION TITLE
Dynamic DataTable Column

### DISCUSSION THREAD

**Discussion by connor-devuk** [primefaces/community#754] [PrimeVue]
Hello so I was wondering if there is a way to create Dynamic columns without actually knowing the columns

e.g something like this
```
<DataTable :value="props.server" paginator :rows="10" tableStyle="min-width: 50rem">
        <Column v-for="(item, index) in Object.entries(props.server)" :key="item[0]" :field="item[1]" :header="index"></Column>
    </DataTable>

```
my props.server array is always different meaning sometimes it might have an id field sometimes it might now, is there a way to achieve this?

Thanks

**Comment by tugcekucukoglu** [comment]
`field` property is one of the musts for the Column component. You can use Columns dynamically after making other necessary adjustments.

**Reply by connor-devuk** [reply]
Could you show me an example of how I would dynamically have each column from the current row

```
<DataTable :value="fixProperties(slotProps.data.properties)">
                    <Column v-for="(col, index) of currentRow" :key="col"> // something like this but idk how to get the currentRow
                        
                    </Column>
                </DataTable>
```
